### PR TITLE
Implement date control on duration dropdown (mac)

### LINKED ISSUE #4611: Add calendar support to Duration Dropdown
### 💻 Environment
Platform: macOS

### 📒 Description
Clicking on the middle of Day Control should show Calendar (embedded into the dropdown view, like on the design).
Changing the date with the Day Control and Calendar should change the time entry start date.

### PR DESCRIPTION
**Pull request by AndrewVebster**
### 📒 Description
Adds working date control to duration dropdown along with an embedded calendar.

### 🕶️ Types of changes

- **New feature** (non-breaking change which adds functionality)

### 👫 Relationships

Closes #4611

### 🔎 Review hints

Duration dropdown is hidden under local feature flag. To enable it go to `TimerDurationControl.swift` line 28 and set `isDurationDropdownEnabled` to `true`

### CODE REVIEW

**Comment by AndrewVebster** [comment]
Saw that changing date won't round duration to minutes. Will try to do the same as with start time. Sorry, do not review for now

**Comment by AndrewVebster** [comment]
Updated, ready for review again

**Comment by AndrewVebster** [comment]
Will make PR for keyboard navigation and analytics right after this one merged

**Review by skel35 — CHANGES_REQUESTED** [review]
Bug discovered:
STR
1.Timer is not running
2. Set the time to 30 mins after the current time

AR
Crash
image

I've also noticed that opening the calendar takes a long time (~1sec). Do you think there's anything that can be done to make it faster?

**Comment by AndrewVebster** [comment]
thanks for catching this, updated PR

File: src/ui/osx/TogglDesktop/Features/Timer/TimerViewController/TimerViewModel.swift [hunk 1]
```diff
@@ -232,14 +231,50 @@ final class TimerViewModel: NSObject {
             newTimestamp += TimeInterval(Date().seconds)
         }
 
-        timeEntry.started = Date(timeIntervalSince1970: newTimestamp)
-        startTimeString = timeString(fromDate: timeEntry.started)
+        var newDate = Date(timeIntervalSince1970: newTimestamp)
+        // `newDay` is always "today" or "yesterday"
```
[the remark this answers is not in the document]
Comment (AndrewVebster) [reply src/ui/osx/TogglDesktop/Features/Timer/TimerViewController/TimerViewModel.swift]: updated, thanks

**Review by skel35 — APPROVED** [review]
couldn't find any more bugs, works just as I'd expect it 👍

**Pull request merged**